src/test/fuzz/transaction.cpp

```diff
@@ -28,6 +30,26 @@ void initialize()
 
 void test_one_input(const std::vector<uint8_t>& buffer)
 {
+    {
+        FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+        const CScript script = ConsumeScript(fuzzed_data_provider);
+        const Optional<COutPoint> out_point = ConsumeDeserializable<COutPoint>(fuzzed_data_provider);
+        if (out_point) {
+            const CTxIn tx_in{*out_point, script, fuzzed_data_provider.ConsumeIntegral<uint32_t>()};
+            (void)tx_in;
+        }
+        const CTxOut tx_out_1{ConsumeMoney(fuzzed_data_provider), script};
+        const CTxOut tx_out_2{ConsumeMoney(fuzzed_data_provider), ConsumeScript(fuzzed_data_provider)};
+        assert((tx_out_1 == tx_out_2) != (tx_out_1 != tx_out_2));
+        const Optional<CMutableTransaction> mutable_tx_1 = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+        const Optional<CMutableTransaction> mutable_tx_2 = ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider);
+        if (mutable_tx_1 && mutable_tx_2) {
+            const CTransaction tx_1{*mutable_tx_1};
+            const CTransaction tx_2{*mutable_tx_2};
+            assert((tx_1 == tx_2) != (tx_1 != tx_2));
+        }
+    }
```

Comment:
It still seems odd. How do you know that the fuzzer won't favor inputs that start with a valid transaction and is more likely to discard inputs that do not?

Reply:
I see your point: yes, it could confuse fuzzers. To rule out that possibility I've now moved this to a new fuzzing harness file :)